src/hooks/screen/useChatScreenHook.ts

```diff
@@ -180,48 +195,78 @@ function useChatScreenHook(type: 'SIGNED' | 'ANONYMOUS'): UseChatScreenHook {
+            selectedChannel?.id || '',
+            message,
+            attachments,
+            localDb,
+            'regular',
+            'pending'
+          );
+
+          await currentChatSchema?.save(localDb);
+          initChatData();
+        } else {
           queue.addPriorityJob({
-            operationLabel: DatabaseOperationLabel.ChatScreen_UpdateChannelDescription,
-            id: `${selectedChannel?.id}-${randomId}`,
+            operationLabel: DatabaseOperationLabel.ChatScreen_SendChat,
+            id: `${selectedChannel?.id}-${new Date().valueOf()}`,
             priority: QueueJobPriority.HIGH,
+            forceAddToQueue: true,
             task: async () => {
-              const channelList: ChannelList | null = await ChannelList.getSchemaById(
+              currentChatSchema = await ChatSchema.generateSendingChat(
+                randomId,
+                userId,
+                selectedChannel?.id || '',
+                message,
+                attachments,
                 localDb,
-                selectedChannel?.id
+                'regular',
+                'pending'
               );
-              if (channelList) {
-                channelList.description = message;
-                channelList.lastUpdatedBy = userId;
-                channelList.lastUpdatedAt = new Date().toISOString();
-                await channelList.save(localDb);
-                refresh('channelList');
-              }
+
+              await currentChatSchema?.save(localDb);
+              initChatData();
             }
           });
         }
+
+        if (selectedChannel) {
+          if (Platform.OS === 'android') {
+            const channelList: ChannelList | null = await ChannelList.getSchemaById(
+              localDb,
+              selectedChannel?.id
+            );
+            if (channelList) {
+              channelList.description = message;
+              channelList.lastUpdatedBy = userId;
+              channelList.lastUpdatedAt = new Date().toISOString();
+              await channelList.save(localDb);
+              refresh('channelList');
+            }
+          } else {
+            queue.addPriorityJob({
+              operationLabel: DatabaseOperationLabel.ChatScreen_UpdateChannelDescription,
+              id: `${selectedChannel?.id}-${randomId}`,
+              priority: QueueJobPriority.HIGH,
+              task: async () => {
+                const channelList: ChannelList | null = await ChannelList.getSchemaById(
+                  localDb,
+                  selectedChannel?.id
+                );
+                if (channelList) {
+                  channelList.description = message;
+                  channelList.lastUpdatedBy = userId;
+                  channelList.lastUpdatedAt = new Date().toISOString();
+                  await channelList.save(localDb);
+                  refresh('channelList');
+                }
+              }
+            });
+          }
+        }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similar to the previous change, this block of code introduces platform-specific logic for Android and other platforms. The same considerations apply here: ensure that these operations can be executed independently without causing race conditions or inconsistencies in the application state across different platforms.
```diff
+      if (Platform.OS === 'android') {
+        currentChatSchema = await ChatSchema.generateSendingChat(
+          randomId,
+          userId,
+          selectedChannel?.id || '',
+          message,
+          attachments,
+          localDb,
+          'regular',
+          'pending'
+        );
+
+        await currentChatSchema?.save(localDb);
+        initChatData();
+      } else {
+         queue.addPriorityJob({
+          operationLabel: DatabaseOperationLabel.ChatScreen_SendChat,
+          id: `${selectedChannel?.id}-${new Date().valueOf()}`,
+           priority: QueueJobPriority.HIGH,
+          forceAddToQueue: true,
+           task: async () => {
+            currentChatSchema = await ChatSchema.generateSendingChat(
+              randomId,
+              userId,
+              selectedChannel?.id || '',
+              message,
+              attachments,
+               localDb,
+              'regular',
+              'pending'
+             );
+
+            await currentChatSchema?.save(localDb);
+            initChatData();
+           }
+         });
+       }
+
+      if (selectedChannel) {
+        if (Platform.OS === 'android') {
+          const channelList: ChannelList | null = await ChannelList.getSchemaById(
+            localDb,
+            selectedChannel?.id
+          );
+          if (channelList) {
+            channelList.description = message;
+            channelList.lastUpdatedBy = userId;
+            channelList.lastUpdatedAt = new Date().toISOString();
+            await channelList.save(localDb);
+            refresh('channelList');
+          }
+        } else {
+          queue.addPriorityJob({
+            operationLabel: DatabaseOperationLabel.ChatScreen_UpdateChannelDescription,
+            id: `${selectedChannel?.id}-${randomId}`,
+            priority: QueueJobPriority.HIGH,
+            task: async () => {
+              const channelList: ChannelList | null = await ChannelList.getSchemaById(
+                localDb,
+                selectedChannel?.id
+              );
+              if (channelList) {
+                channelList.description = message;
+                channelList.lastUpdatedBy = userId;
+                channelList.lastUpdatedAt = new Date().toISOString();
+                await channelList.save(localDb);
+                refresh('channelList');
+              }
+            }
+          });
+        }
+      }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->